app/client/src/store/actions.ts

```diff
@@ -47,54 +60,33 @@ export default {
     Object.keys(state.results.perIsolate).forEach((element) => {
       jsonSketches[element] = JSON.parse(state.results.perIsolate[element].sketch as string);
     });
-    await axios.post(
-      `${config.server_url}/poppunk`,
-      { projectHash: phash, sketches: jsonSketches },
-      {
-        headers: {
-          'Content-Type': 'application/json',
-        },
-      },
-    ).then(() => {
-      commit('setStatus', { task: 'assign', data: 'submitted' });
-      commit('setStatus', { task: 'microreact', data: 'submitted' });
-      commit('setStatus', { task: 'network', data: 'submitted' });
-    });
+    await axiosJsonPost('poppunk', { projectHash: phash, sketches: jsonSketches })
+      .then(() => {
+        commit('setAnalysisStatus', { assign: 'submitted', microreact: 'submitted', network: 'submitted' });
+      });
   },
-  async getStatus({ commit, state }: { commit: Commit, state: RootState }) {
-    await axios.post(
-      `${config.server_url}/status`,
-      { hash: state.projectHash },
-      {
-        headers: {
-          'Content-Type': 'application/json',
-        },
-      },
-    ).then((response) => {
-      const prev = { ...state.analysisStatus };
-      if (response.data.data.microreact !== prev.microreact) {
-        commit('setStatus', { task: 'microreact', data: response.data.data.microreact });
-      }
-      if (response.data.data.network !== prev.network) {
-        commit('setStatus', { task: 'network', data: response.data.data.network });
-      }
-      if (response.data.data.assign !== prev.assign) {
-        commit('setStatus', { task: 'assign', data: response.data.data.assign });
-      }
-    });
+  async getStatus(context: ActionContext<RootState, RootState>) {
+    const { state, dispatch, commit } = context;
+    await axiosJsonPost('status', { hash: state.projectHash })
+      .then((response) => {
+        if (response.data.data.assign === 'finished' && state.analysisStatus.assign !== 'finished') {
+          dispatch('getAssignResult');
+        }
+        commit('setAnalysisStatus', response.data.data);
+        if ((response.data.data.network === 'finished' || response.data.data.network === 'failed') && (response.data.data.microreact === 'finished' || response.data.data.microreact === 'failed')) {
```

Comment:
I'm seeing status requests continue to fire even when microreact and network are showing as 'finished'. I think that's probably because I tested if I could still click the 'Start Analysis' button after it had started running, so probably the first interval is still firing and only the second one got cleared.

Update: yes, I tried again and just clicked the button once, and it stopped polling when expected.